sdk/identity/identity/src/credentials/clientSecretCredential.ts

```diff
@@ -58,49 +57,10 @@ export class ClientSecretCredential implements TokenCredential {
    * @param options - The options used to configure any requests this
    *                TokenCredential implementation might make.
    */
-  public async getToken(
-    scopes: string | string[],
-    options?: GetTokenOptions
-  ): Promise<AccessToken | null> {
-    const { span, updatedOptions } = createSpan("ClientSecretCredential-getToken", options);
-    try {
-      const urlSuffix = getIdentityTokenEndpointSuffix(this.tenantId);
-      const webResource = this.identityClient.createWebResource({
-        url: `${this.identityClient.authorityHost}/${this.tenantId}/${urlSuffix}`,
-        method: "POST",
-        disableJsonStringifyOnBody: true,
-        deserializationMapper: undefined,
-        body: qs.stringify({
-          response_type: "token",
-          grant_type: "client_credentials",
-          client_id: this.clientId,
-          client_secret: this.clientSecret,
-          scope: typeof scopes === "string" ? scopes : scopes.join(" ")
-        }),
-        headers: {
-          Accept: "application/json",
-          "Content-Type": "application/x-www-form-urlencoded"
-        },
-        abortSignal: options && options.abortSignal,
-        spanOptions: updatedOptions?.tracingOptions?.spanOptions
-      });
-
-      const tokenResponse = await this.identityClient.sendTokenRequest(webResource);
-      logger.getToken.info(formatSuccess(scopes));
-      return (tokenResponse && tokenResponse.accessToken) || null;
-    } catch (err) {
-      const code =
-        err.name === AuthenticationErrorName
-          ? CanonicalCode.UNAUTHENTICATED
-          : CanonicalCode.UNKNOWN;
-      span.setStatus({
-        code,
-        message: err.message
-      });
-      logger.getToken.info(formatError(scopes, err));
-      throw err;
-    } finally {
-      span.end();
-    }
+  async getToken(scopes: string | string[], options: GetTokenOptions = {}): Promise<AccessToken> {
+    return trace(`${this.constructor.name}.getToken`, options, async (newOptions) => {
+      const arrayScopes = typeof scopes === "object" ? scopes : [scopes];
```

Comment:
if it wasn't a breaking change, I'd love to make the contract just `scopes: string[]`

Also, technically, you're supposed to use `Array.isArray` rather than type tests

Reply:
I'll use Array.isArray!